panic: runtime error: index out of range [-3352]

I have installed ZDNS on raspberry pi 3 model B v1.2 and build using Go v1.24.5, the tool works fine with some domains like (google.com) while resolving domains such as (archive.org) causes the following error to show up:
-----------------------------------
panic: runtime error: index out of range [-3352]

goroutine 39 [running]:
github.com/zmap/zdns/v2/src/internal/cachehash.(*ShardedCacheHash).getShard(...)
	/home/pi/zdns/src/internal/cachehash/shardedcachehash.go:42
github.com/zmap/zdns/v2/src/internal/cachehash.(*ShardedCacheHash).Lock(0x17a25c0, {0x6b7a30, 0x1852c90})
	/home/pi/zdns/src/internal/cachehash/shardedcachehash.go:72 +0x68
github.com/zmap/zdns/v2/src/zdns.(*Cache).getCachedResult(0x17a25c0, {0x1, 0x1, {0x14a3160, 0xf}}, 0x1852c60, 0x0, 0x2)
	/home/pi/zdns/src/zdns/cache.go:123 +0x5e4
github.com/zmap/zdns/v2/src/zdns.(*Cache).GetCachedResults(...)
	/home/pi/zdns/src/zdns/cache.go:97
github.com/zmap/zdns/v2/src/zdns.(*Resolver).cachedLookup(0x1858008, {0x824538, 0x1797c70}, {0x1, 0x1, {0x14a3160, 0xf}}, 0x1852c60, {0x14a3160, 0xf}, ...)
	/home/pi/zdns/src/zdns/lookup.go:857 +0x3f8
github.com/zmap/zdns/v2/src/zdns.(*Resolver).cyclingLookup(0x1858008, {0x824538, 0x1797c70}, 0x14aaba0, {0x1852c60, 0x1, 0x1}, {0x14a3160, 0xf}, 0x1, ...)
	/home/pi/zdns/src/zdns/lookup.go:783 +0x220
github.com/zmap/zdns/v2/src/zdns.(*Resolver).lookup(0x1858008, {0x824538, 0x1797c70}, 0x14aaba0, {0x1852c60, 0x1, 0x1}, 0x0, {0x0, 0x0, ...})
	/home/pi/zdns/src/zdns/lookup.go:174 +0x548
github.com/zmap/zdns/v2/src/zdns.(*Resolver).followingLookup(0x1858008, {0x824538, 0x1797c70}, 0x14aaba0, {0x1852c60, 0x1, 0x1}, 0x0)
	/home/pi/zdns/src/zdns/lookup.go:217 +0x328
github.com/zmap/zdns/v2/src/zdns.(*Resolver).doDstServersLookup(0x1858008, {0x824538, 0x1797c70}, {0x1, 0x1, {0x14a3160, 0xf}}, {0x1852c60, 0x1, 0x1}, ...)
	/home/pi/zdns/src/zdns/lookup.go:145 +0x3b8
github.com/zmap/zdns/v2/src/zdns.LookupClient.DoDstServersLookup({}, {0x824538, 0x1797c70}, 0x1858008, {0x1, 0x1, {0x14a3160, 0xf}}, {0x1852c60, 0x1, ...}, ...)
	/home/pi/zdns/src/zdns/lookup.go:108 +0x6c
github.com/zmap/zdns/v2/src/zdns.(*Resolver).ExternalLookup(0x1858008, {0x82451c, 0x14ec600}, 0x144fd14, 0x0)
	/home/pi/zdns/src/zdns/resolver.go:640 +0x320
github.com/zmap/zdns/v2/src/cli.(*BasicLookupModule).Lookup(0x14aa500, {0x82451c, 0x14ec600}, 0x1858008, {0x14a3160, 0xf}, 0x0)
	/home/pi/zdns/src/cli/modules.go:196 +0x2fc
github.com/zmap/zdns/v2/src/cli.handleWorkerInput({0x82451c, 0x14ec600}, 0x17a6008, 0x179fba8, {0x7eec0577, 0xf}, 0x1858008, 0x1479f60, 0x1490e80, 0x1490ec0)
	/home/pi/zdns/src/cli/worker_manager.go:709 +0x5ec
github.com/zmap/zdns/v2/src/cli.doLookupWorker({0x82451c, 0x14ec600}, 0x17a6008, 0x179fba8, 0x1490e40, 0x1490e80, 0x17a2680, 0x1490ec0, 0x171a010)
	/home/pi/zdns/src/cli/worker_manager.go:646 +0x18c
github.com/zmap/zdns/v2/src/cli.Run.func5(0x1)
	/home/pi/zdns/src/cli/worker_manager.go:575 +0x64
created by github.com/zmap/zdns/v2/src/cli.Run in goroutine 1
	/home/pi/zdns/src/cli/worker_manager.go:574 +0x758
pi@raspberrypi:~/zdns $

--- a/src/internal/cachehash/shardedcachehash.go
+++ b/src/internal/cachehash/shardedcachehash.go
@@ -35,7 +35,13 @@ func (c *ShardedCacheHash) Init(maxLen int, shards int) {
 
 func (c *ShardedCacheHash) getShardID(k interface{}) int {
 	kb := []byte(fmt.Sprintf("%v", k))
-	return int(crc32.ChecksumIEEE(kb)) % c.shardsLen
+	hash := int(crc32.ChecksumIEEE(kb))
+	// crc32 returns a 32-bit unsigned integer. On 32-bit systems, this int cast can return negative values.
+	// To ensure we always have 0 <= shard ID < shardsLen, we take the absolute value of the hash.
+	if hash < 0 {
+		hash = -hash
+	}
+	return hash % c.shardsLen
 }
 
 func (c *ShardedCacheHash) getShard(k interface{}) *CacheHash {

